acvm/src/lib.rs

```diff
@@ -74,24 +74,24 @@ pub trait PartialWitnessGenerator {
     fn schnorr_verify(
         &self,
         initial_witness: &mut WitnessMap,
-        public_key_x: &FunctionInput,
-        public_key_y: &FunctionInput,
-        signature: &[FunctionInput],
+        public_key_x: FunctionInput,
+        public_key_y: FunctionInput,
+        signature: (FunctionInput, FunctionInput),
```

Comment:
If you want, I'd prefer `sig_s` and `sig_e` just because you can provide a bit more context versus the tuple where its not clear what the first item and second item are. It also would be harder to mess up named fields version structs.

This can ofcourse be aleviated with proper documentation, so no strong incling if you prefer the other